jwala-services/src/main/java/com/cerner/jwala/service/jvm/state/JvmStateReceiverAdapter.java

```diff
@@ -69,18 +70,73 @@ public void receive(final Message jGroupMsg) {
         }
     }
 
+    private JvmState getJvmState(final Map serverInfoMap) {
+
+        // first check if the key types are string
+        final Set keys = serverInfoMap.keySet();
+        if (keys.isEmpty()) {
+            return null;
+        }
+
+        final Object initialKey = keys.iterator().next();
+        if (initialKey instanceof String) {
```

Comment:
Yes I understand but I was thinking the key "STATE" will have a different hash code to a key of type ReportingJmsMessageKey. I actually tried it. Anyways, we can also go with checking on the type but can we make it consistent with the other method e.g. check instance of ReportingJmsMessageKey instead of String ? Then we can ditch the "assume the key is...JvmState".